proto/tendermint/abci/types.proto

```diff
@@ -196,6 +196,10 @@ message ResponseCheckTx {
   string         codespace  = 8;
   string         sender     = 9;
   int64          priority   = 10;
+
+  // mempool_error is set by Tendermint.
+  // ABCI applictions creating a ResponseCheckTX should not set mempool_error.
```

Comment:
```suggestion
  // ABCI applications creating a ResponseCheckTX should not set mempool_error.
```

Reply:
Also what's the context for adding this field if it isn't set by the application?

Reply:
https://github.com/tendermint/tendermint/pull/6740